src/index.ts

```diff
@@ -68,6 +70,15 @@ export interface WorkspacePermissionsSpec {
   public: boolean;
 }
 
+export interface WorkspacePermissionRequestSpec {
+  /* In order to set permissions, the API only needs usernames */
+  owner: MinimumUserSpec;
+  maintainers: MinimumUserSpec[];
+  writers: MinimumUserSpec[];
+  readers: MinimumUserSpec[];
```

Comment:
If this is the only place `MinimumUserSpec` is needed, then it seems better to use `string` in place of it, and "promote" the strings into the proper structure needed by the endpoint after the data arrives inside the appropriate function. This is in the interest of keeping the calling sequences for the functions in question as human-centric and intuitive as possible.

If `MinimumUserSpec` is unavoidable, then I'd refactor its definition and that of `UserSpec` to use an `extends` relationship.

Reply:
Turns out that `MinimumUserSpec` is not needed. The most recent commit removes it.